widgets/bat_openbsd.lua

```diff
@@ -0,0 +1,72 @@
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local tonumber = tonumber
+local string = { format = string.format, gmatch = string.gmatch }
+local io = { popen = io.popen }
+local math = { floor = math.floor }
+local helpers = require("vicious.helpers")
+local string = {
+    gmatch = string.gmatch,
+    match = string.match,
+    format = string.format
+}
+-- }}}
+local bat_openbsd = {}
+
+local function worker(format, warg)
+    local battery = warg or "bat0"
+    local filter = string.format("hw.sensors.acpi%s", battery)
+    local cmd = string.format("sysctl -a | grep '^%s'", filter)
+    local proc = io.popen(cmd)
+
+    local bat_info = {}
+    for line in proc:lines("*l") do
+        for key, value in string.gmatch(line, "(%S+)=(%S+)") do
+	    key = key:gsub(filter .. ".", "")
+	    bat_info[key] = value
+        end
+    end
+
+    -- current state
+    local states = {[0] = "↯", -- not charging
+                    [1] = "-", -- discharging
+                    [2] = "!", -- critical
+                    [3] = "+", -- charging
+                    [4] = "N/A", -- unknown status
+                    [255] = "N/A"} -- unimplemented by the driver
+    local state = states[tonumber(bat_info.raw0)]
+
+    -- battery capacity in percent
+    local percent = tonumber(bat_info.watthour3 / bat_info.watthour0 * 100)
+
+    local time
+    if tonumber(bat_info.power0) < 1 then
+        time = "∞"
+    else
+	local raw_time = bat_info.watthour3 / bat_info.power0
+	local hours = math.floor(raw_time)
+	local minutes = raw_time % 1
+	time = string.format("%d:%0.2d", hours, minutes)
```

Comment:
Relating to https://github.com/vicious-widgets/vicious/pull/75#issuecomment-471214820, could you please also floor minutes for Lua 5.3 compatibility? Or you could explain why is `minutes = raw_time % 1` so that I have to confidence to do the change myself.

Reply:
Thanks again for your feedback. I've now addressed your comments in #78. Line 48 shoud've read `... 60 * (raw_time % 1). That is fixed in #78, along with some cruft that was detected with `luacheck`.